admin/listPurchases/listView.html

```diff
@@ -120,73 +131,86 @@
             e.preventDefault();
-            if(entryPageIndex >= amountOfSubpages - 1) return;
-            entryPageIndex++;
+            if(currentPageIndex >= amountOfSubpages - 1) return;
+            currentPageIndex++;
             updateContent();
         });
         document.querySelector(`.subpage-selector-to-start`).addEventListener("click", function (e){
             e.preventDefault();
-            entryPageIndex = 0;
+            currentPageIndex = 0;
             updateContent();
         });
         document.querySelector(`.subpage-selector-to-end`).addEventListener("click", function (e){
             e.preventDefault();
-            entryPageIndex = amountOfSubpages - 1;
+            currentPageIndex = amountOfSubpages - 1;
             updateContent();
         });
     }
     
     for(let i = startIndexToRender; i < startIndexToRender + amountOfShownSubpages && i < amountOfSubpages; i++){
         document.querySelector(`.subpage-selector-index-${i}`).addEventListener("click", function (e){
             e.preventDefault();
-            entryPageIndex = i;
+            currentPageIndex = i;
             updateContent();
         });
-        
     }
 
-    if(amountOfSubpages) document.querySelector(`.subpage-selector-index-${entryPageIndex}`).classList.add("selected");
+    if(amountOfSubpages) {
+        document.querySelector(`.subpage-selector-index-${currentPageIndex}`).classList.add("selected");
+    }
+
     updateSubpageInputBox();
 }
 
 const inputEle = document.getElementById("subpageInputBox");
 
 function updateSubpageInputBox(){
     inputEle.setAttribute("max", amountOfSubpages);
-    inputEle.value = entryPageIndex + 1;
+    inputEle.value = currentPageIndex + 1;
     document.getElementById("subpageInputLabel").innerText = "ud af " + amountOfSubpages;
 }
 
 inputEle.addEventListener("change", () => {
     if(inputEle.value > amountOfSubpages) inputEle.value = amountOfSubpages;
     if(inputEle.value < 1) inputEle.value = 1;
-    entryPageIndex = inputEle.value - 1;
+    currentPageIndex = inputEle.value - 1;
     updateContent();
 });
 
 const orderSearchBar = document.getElementById("order-search-bar");
-var searchParams;
-orderSearchBar.addEventListener("change", () =>{entryPageIndex = 0; searchParams = orderSearchBar.value.split(" "); updateContent();});
+let searchParams;
+orderSearchBar.addEventListener("change", () =>{currentPageIndex = 0; searchParams = orderSearchBar.value.split(" "); updateContent();});
 
 function updateContent(){
+    const entriesFiltered = entries.filter(currentOption.filter);
     if(!orderSearchBar.value){
-        renderEntriesToList(entries.filter(currentOption.filter).slice(entryPageIndex * entriesPerSubpage, entryPageIndex * entriesPerSubpage + entriesPerSubpage));
+        const entriesFilteredSectioned = entriesFiltered.slice(currentPageIndex * entriesPerSubpage, currentPageIndex * entriesPerSubpage + entriesPerSubpage);
```

Comment:
You do a lot of repeated calculations. Whenever you do these, you should extract them. They are likely to change at the same time in the future:

```js
const entryOffset = currentPageIndex * entriesPerSubpage;
const entriesFilteredSectioned = entriesFiltered.slice(entryOffset, entryOffset + entriesPerSubpage);
```

This code is also much more readable because the concept of an entry offset is made clear